Modules/Tracker/QuestieTracker.lua

```diff
@@ -1087,8 +1808,115 @@ _AQW_Insert = function(index, expire)
                 Questie.db.char.AutoUntrackedQuests[qid] = true
             end
         end
-        C_Timer.After(0.1, function()
-            QuestieTracker:Update()
-        end)
+
+        -- Make sure quests or zones (re)added to the tracker isn't in a minimized state
+        local quest = QuestieDB:GetQuest(qid)
+        local zoneId = quest.zoneOrSort
+
+        if Questie.db.char.collapsedQuests[qid] == true then
+            Questie.db.char.collapsedQuests[qid] = nil
+        end
+
+        if Questie.db.char.collapsedZones[zoneId] == true then
+            Questie.db.char.collapsedZones[zoneId] = nil
+        end
+
+        QuestieTracker:ResetLinesForChange()
+        QuestieTracker:Update()
+    end
+end
+
+local function getWorldPlayerPosition()
+    -- Turns coords into 'world' coords so it can be compared with any coords in another zone
+    local uiMapId = C_Map.GetBestMapForUnit("player");
+
+    if not uiMapId then
+        return nil;
+    end
+
+    local mapPosition = C_Map.GetPlayerMapPosition(uiMapId, "player");
+    local _, worldPosition = C_Map.GetWorldPosFromMapPos(uiMapId, mapPosition);
+
+    return worldPosition;
+end
+
+local function getDistance(x1, y1, x2, y2)
+    -- Basic proximity distance calculation to compare two locs (normally player position and provided loc)
+    return math.sqrt( (x2-x1)^2 + (y2-y1)^2 );
+end
+
+function GetDistanceToClosestObjective(questId)
```

Comment:
If this is a global function please add it to a module. If not then make it local with an underscore prefix